tests/test_logic.py

```
@@ -0,0 +1,89 @@
+def brightness_steps() -> list[int]:
+    """Return a sample list of brightness steps."""
+    return [10, 25, 50, 100]
+
+
+def test_initialization(brightness_steps: list[int]) -> None:
+    """Test the initialization of the StepDimmerLogic."""
+    logic = StepDimmerLogic(brightness_steps)
+    assert logic._brightness_steps == [10, 25, 50, 100]
+    assert logic._num_steps == 4
+
+    with pytest.raises(ValueError):
```

Ruff (0.14.3)

`pytest.raises(ValueError)` is too broad, set the `match` parameter or use a more specific exception

(PT011)